File: problems/p020.js
var bigInt = require('../utils/bigInt');

var nums = [];
var result = 0;
var factorial;
var i;

for (i = 2; i <= 100; i++) {
    nums.push(i.toString());
}

factorial = bigInt.mul.apply(bigInt, nums);

for (i = 0; i < factorial.length; i++) {
    result += parseInt(factorial[i], 10);
}

console.log(result);

File: rust/p020.rs
mod big_int;

fn main() {
    let mut fact = big_int::from_u32(1);

    for i in 2..=100 {
        fact = big_int::mul(&fact, &big_int::from_u32(i));
    }

    let result = fact.to_string()
        .chars()
        .map(|c| c as i32 - '0' as i32)
        .fold(0, |acc, d| acc + d);

    println!("{}", result);
}
